### PR TITLE
Add bikelane data to api seed, add sf311_case_metadata table, and rem…

### PR DESCRIPTION
**Pull request by sahilda**
…ove mobile_metadata table

@alex-wood

### CODE REVIEW

**Comment by sahilda** [comment]
Hey @alex-wood, take a look at the `Sf311Case` `after_commit` hook I added to add the metadata and see if it makes sense. In order to support the hook during dev seeding, I had to move away from the `import` function and also decreased seeding to only add 50 records to keep it quick still.

File: lanebreach-api/app/models/sf311_case.rb [hunk 1]
```diff
@@ -32,9 +32,17 @@
 #
 
 class Sf311Case < ApplicationRecord
+  has_one :sf311_case_metadatum
+
+  after_commit :add_metadata
```
Comment (alex-wood) [inline lanebreach-api/app/models/sf311_case.rb]: I think we may want to change this to an `after_create` callback so `Sf311CaseMetadatum` aren't created when existing `Sf311Case` records are changed. Thoughts?

Comment (alex-wood) [reply lanebreach-api/app/models/sf311_case.rb]: Additionally, we may want to consider adding a condition to this callback that specifically checks whether the `lat` or `long` attributes have changed.

File: lanebreach-api/app/models/sf311_case_metadatum.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+#  fk_rails_...  (sf311_case_id => sf311_cases.id)
+#
+
+class Sf311CaseMetadatum < ApplicationRecord
+  belongs_to :sf311_case
+
+  def self.create_metadata(sf311_case)
+    bikeway_network = BikewayNetwork.nearest(sf311_case.lat, sf311_case.long)
+    if bikeway_network.present?
+      Sf311CaseMetadatum.create!(:bike_lane_type => bikeway_network.symbology, :sf311_case => sf311_case)
```
Comment (alex-wood) [inline lanebreach-api/app/models/sf311_case_metadatum.rb]: This is nitpicky, but I think we should standardize on Ruby 2 hash syntax (i.e., `bike_lane_type: bikeway_network.symbology`)  to keep code style consistent.

File: lanebreach-api/app/services/sf311_case_service.rb [hunk 3]
```diff
@@ -11,9 +11,8 @@ module Sf311CaseService
   }
 
   CASE_DATASET_ID = 'ktji-gk7t'
-
   # https://dev.socrata.com/docs/datatypes/floating_timestamp.html
-  SODA_FLOATING_TIMESTAMP_FORMAT ='%Y-%m-%dT%T'
+  SODA_FLOATING_TIMESTAMP_FORMAT = '%Y-%m-%dT%T'
```
Comment (alex-wood) [inline lanebreach-api/app/services/sf311_case_service.rb]: Were these changes intentional?

File: lanebreach-api/db/migrate/20181129050151_create_sf311_case_metadata_table.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class CreateSf311CaseMetadataTable < ActiveRecord::Migration[5.2]
+  def change
+    create_table :sf311_case_metadata do |t|
```
Comment (alex-wood) [inline lanebreach-api/db/migrate/20181129050151_create_sf311_case_metadata_table.rb]: What are your thoughts on simply adding `bike_lane_type` and other metadata fields to the `sf_311_case` schema instead of creating an additional table?

File: lanebreach-api/app/services/sf311_case_service.rb (hunk 3 above)
Comment (sahilda) [reply lanebreach-api/app/services/sf311_case_service.rb]: Ah,  yeah didn't mean to check these in

File: lanebreach-api/app/models/sf311_case_metadatum.rb (hunk 2 above)
Comment (sahilda) [reply lanebreach-api/app/models/sf311_case_metadatum.rb]: Totally, good call. I imagine eventually we could add a linter too that can make these checks.

File: lanebreach-api/app/models/sf311_case.rb (hunk 1 above)
Comment (sahilda) [reply lanebreach-api/app/models/sf311_case.rb]: That makes sense about the `after_create` - I'll instead fix the callback to either a create new record or update an existing one.

I'm not expecting a `Sf311Case` object to get mutated after initial creation, so not sure if we need to add the `lat` and `long` _today_ checks but they make sense.

File: lanebreach-api/db/migrate/20181129050151_create_sf311_case_metadata_table.rb (hunk 4 above)
Comment (sahilda) [reply lanebreach-api/db/migrate/20181129050151_create_sf311_case_metadata_table.rb]: No strong preference either way. My thought process was to keep it separate, treating the `sf311_case` data as basically immutable + replica of the import source, and then add our data elsewhere.

This table's still a work-in-progress (like just finding out using `barrier` or `facility_t` is better than `bike_lane_type), and actually just added a reference to the `bikeway_network` table too. So will keep it separate for now as I think it'll be easier to iterate, but really not a strong preference.

Comment (alex-wood) [reply lanebreach-api/db/migrate/20181129050151_create_sf311_case_metadata_table.rb]: Got it, sounds good.
